lambdas/service/app.py

```diff
@@ -974,25 +975,32 @@ def repository_search(entity_type: str,
     validate_repository_search(query_params)
     catalog = app.catalog
     filters = query_params.get('filters')
-    source_ids = {
-        source['sourceId']
-        for source in app.repository_controller.list_sources(catalog, request)
-    } if filter_sources else None
     try:
         service = IndexQueryService()
         return service.get_data(catalog=catalog,
                                 entity_type=entity_type,
                                 file_url_func=app.file_url,
                                 item_id=item_id,
                                 filters=filters,
-                                source_ids=source_ids,
+                                source_filter=source_filter(indicate_only=not filter_sources),
                                 pagination=app.get_pagination(entity_type))
     except (BadArgumentException, InvalidUUIDError) as e:
         raise BadRequestError(msg=e)
     except (EntityNotFoundError, IndexNotFoundError) as e:
         raise NotFoundError(msg=e)
 
 
+def source_filter(*, indicate_only: bool) -> SourceFilter:
+    source_ids = {
+        source['sourceId']
+        for source in app.repository_controller.list_sources(app.catalog, app.current_request)
+    }
+    return SourceFilter(source_ids=source_ids,
+                        indicate_only=indicate_only,
+                        # FIXME clarify intended behavior for entities with multiple sources
```

Comment:
FIXME comments generally should include a `https://github.com/DataBiosphere/azul/issues/####` URL for the issue that will resolve the fixme.

Reply:
This was meant to be a note to myself to ask about this during parking lot it is now resolved.